### PR TITLE
Add validation for update flags

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Fixes https://github.com/giantswarm/inago/issues/105

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
@giantswarm/cena RFR

I've tested this manually, and the validation seems to work. I've opened https://github.com/giantswarm/inago/issues/131 to address the lack of unit tests for update.

File: controller/update.go [hunk 1]
```diff
@@ -201,6 +201,16 @@ func (c controller) UpdateWithStrategy(ctx context.Context, req Request, opts Up
 	var addInProgress int64
 	var removeInProgress int64
 
+	// To keep n units alive during the update,
+	// we need to have just more than that in current units,
+	// and units we're allowed to create during the update.
+	if !(numTotal+opts.MaxGrowth > opts.MinAlive) {
```
Comment (zeisss) [inline controller/update.go:207]: this allows me to do this:

```
inagoctl submit mygroup 5
instalctl start mygroup
inagoctl update mygroup --max-growth=5 --min-alive=9
```

Which clearly means, it cannot reduce the number of slices after the initial growth. I don't think this is what we want.

Comment (JosephSalisbury) [inline controller/update.go:207]: Yeah, this doesn't make sense. I'll have a think.

**Comment by JosephSalisbury** [comment]
I'm closing this for now - https://github.com/giantswarm/inago/pull/126#discussion_r56483602 shows a fairly fundamental issue.

**Pull request closed without merging**